### PR TITLE
Qt: Minimize to tray functionality only on Windows

### PR DESCRIPTION
**Pull request by vajdaz**
Minimizing to tray should only be possible on Windows.
Therefore, on non-Winodws systems the tray icon and related
settings are hidden, the related option variables are set to
predefined values. Furthermore, not used entries are removed
from the configuration files at initialization.
See also discussions in PR #11859

### CODE REVIEW

File: src/qt/optionsmodel.cpp [hunk 1]
```diff
@@ -64,6 +66,15 @@ void OptionsModel::Init(bool resetSettings)
     if (!settings.contains("fMinimizeToTray"))
         settings.setValue("fMinimizeToTray", false);
     fMinimizeToTray = settings.value("fMinimizeToTray").toBool() && !fHideTrayIcon;
+#else
+    // On systems without tray icon functionality we set some default values
+    // and remove settings from previous versions of the software to not
+    // confuse people looking into the .config or .plist file.
+    fHideTrayIcon = true;
+    settings.remove("fHideTrayIcon");
```
Comment (promag) [inline src/qt/optionsmodel.cpp]: IIRC we don't remove old/unused settings. ping @laanwj.

Comment (vajdaz) [reply src/qt/optionsmodel.cpp]: Ok, I have removed the removal of unused settings. To be honest, I realized the "ping @laanwj" only after my commit. Let's hope, Wladimir agrees.

Comment (laanwj) [reply src/qt/optionsmodel.cpp]: Yes I agree, this makes it possible to upgrade/downgrade without losing settings.

**Comment by laanwj** [comment]
utACK, needs testing on Windows/Linux I suppose.

**Comment by kristapsk** [comment]
Why? Minimize to tray works for me on Xfce4.

**Comment by vajdaz** [comment]
On some desktop environments (Xfce4 or KDE for example) minimizing to tray works. Howerver, on other ones (GNOME or Unity for example) there is no system tray. On those desktop environments the user loses visual reference to the application when minimizing to tray (the application "disappears", still running in the background). Linux desktop environments seem not to be universal in the implementation of this feature. Discussions in PR #11859 lead to the idea of removing the "minimize to tray" functionality (at least on non-Windows systems).

**Comment by ryanofsky** [comment]
> Howerver, on other ones (GNOME or Unity for example) there is no system tray.

System tray icon shows up for me with gnome 3.24.2, qt-5.9.3, current master. (It shows up on the bottom left in a little slide out-tray next to hexchat and deluge icons.)

**Comment by vajdaz** [comment]
"Legacy system tray" has been removed in GNOME 3.26. [https://bugzilla.gnome.org/show_bug.cgi?id=785956](https://bugzilla.gnome.org/show_bug.cgi?id=785956)

**Comment by laanwj** [comment]
I've heard of issues with Xfce4 too. The icon appears initially, but seems to randomly disappear. Normally that's not that bad, but it is if it is the only way to get the window back (which is what this option enables).
The systray situation on Linux is very unreliable.

**Comment by kristapsk** [comment]
I'm using it all the time under Xfce4 this way, can't remember having any issues. Bitcoin wallet is not the piece of software I want to pollute my anyway full taskbars with (as it's something that runs in background and I don't need to switch to it so often).

Anyway, I'm Gentoo user, where everything is source based anyway, so, if this patch gets applied, I will anyway add my own extra patch on top of the release sources to get this functionality back.

**Comment by laanwj** [comment]
Looking at the replies, one guess would be that this is mostly a problem with the pre-built binaries, and less when building from source against a Qt5 tuned for the specific environment.

**Comment by ryanofsky** [comment]
It's unclear why this is implemented using a compile time macro rather than calling [QSystemTrayIcon::isSystemTrayAvailable](http://doc.qt.io/qt-5/qsystemtrayicon.html#isSystemTrayAvailable) or something at runtime.

I think if this feature is going to be disabled, there should at least be a proper bug report about where the problem is occurring so it can be revisited in the future. vajdaz seems to be a windows user, not a gnome or unity user. I use gnome and use tray icons and haven't had a problem with them.

**Comment by laanwj** [comment]
Closing this for now, seems to be no agreement to do this.

**Pull request closed without merging**